From fff2396a246949d27212cd9eb035c463e223c79a Mon Sep 17 00:00:00 2001
From: Khanh Luu <khanh.luu.xw@renesas.com>
Date: Thu, 28 Mar 2019 11:45:35 +0700
Subject: [PATCH 027/248] drm: rcar-lvds: Add r8a774c0 device support

Add support for the RZ/G2E (R8A774C0) SoCs to the R-Car LVDS.

Signed-off-by: Khanh Luu <khanh.luu.xw@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_lvds.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/gpu/drm/rcar-du/rcar_lvds.c b/drivers/gpu/drm/rcar-du/rcar_lvds.c
index 4c39de3..178c748 100644
--- a/drivers/gpu/drm/rcar-du/rcar_lvds.c
+++ b/drivers/gpu/drm/rcar-du/rcar_lvds.c
@@ -35,6 +35,10 @@ enum rcar_lvds_mode {
 #define RCAR_LVDS_QUIRK_GEN2_PLLCR (1 << 1)	/* LVDPLLCR has gen2 layout */
 #define RCAR_LVDS_QUIRK_GEN3_LVEN (1 << 2)	/* LVEN bit needs to be set */
 						/* on R8A77970/R8A7799x */
+/* Has extended PLL */
+#define RCAR_LVDS_QUIRK_EXT_PLL		BIT(3)
+/* Supports dual-link operation */
+#define RCAR_LVDS_QUIRK_DUAL_LINK	BIT(4)
 
 struct rcar_lvds_device_info {
 	unsigned int gen;
@@ -514,6 +518,12 @@ static const struct rcar_lvds_device_info rcar_lvds_r8a77970_info = {
 	.quirks = RCAR_LVDS_QUIRK_GEN2_PLLCR | RCAR_LVDS_QUIRK_GEN3_LVEN,
 };
 
+static const struct rcar_lvds_device_info rcar_lvds_r8a774c0_info = {
+	.gen = 3,
+	.quirks = RCAR_LVDS_QUIRK_GEN3_LVEN | RCAR_LVDS_QUIRK_EXT_PLL
+		| RCAR_LVDS_QUIRK_DUAL_LINK,
+};
+
 static const struct of_device_id rcar_lvds_of_table[] = {
 	{ .compatible = "renesas,r8a7743-lvds", .data = &rcar_lvds_gen2_info },
 	{ .compatible = "renesas,r8a7790-lvds", .data = &rcar_lvds_r8a7790_info },
@@ -522,6 +532,8 @@ static const struct of_device_id rcar_lvds_of_table[] = {
 	{ .compatible = "renesas,r8a7795-lvds", .data = &rcar_lvds_gen3_info },
 	{ .compatible = "renesas,r8a7796-lvds", .data = &rcar_lvds_gen3_info },
 	{ .compatible = "renesas,r8a77970-lvds", .data = &rcar_lvds_r8a77970_info },
+	{ .compatible = "renesas,r8a774c0-lvds",
+	.data = &rcar_lvds_r8a774c0_info },
 	{ }
 };
 
-- 
2.7.4

